addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as ts from 'typescript';
+import * as glob from 'glob';
+import * as path from 'path';
+import * as denodeify from 'denodeify';
+
+import { Promise } from 'es6-promise';
+
+/**
+ * Declare an Interface where key is a string and value is either a string or a number.
+ * Use for storing a string literal, start position and end posittion of ImportClause node kinds.
+ */
+export interface ModuleImport {
+  specifierText: string;
+  pos: number;
+  end: number;
+};
+
+export interface ModuleMap {
+  [key: string]: ModuleImport[];
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createTsSourceFile(fileName: string): Promise<ts.SourceFile> {
+  const readFile = denodeify(fs.readFile);
+  return readFile(fileName, 'utf8')
+    .then((contents: string) => {
+      return ts.createSourceFile(fileName, contents, ts.ScriptTarget.ES6, true);
+    });
+}
+
+/**
+ * @return traverses through ts.Node and returns an array of moduleSpecifiers
+ * 
+ * @param {ts.SourceFile} node: Typescript Node of whose AST is being traversed
+ * 
+ */
+export function getImportClauses(node: ts.SourceFile): ModuleImport[] {
+  return node.statements
+    .filter(node => node.kind === ts.SyntaxKind.ImportDeclaration)  // Only Imports.
+    .map((node: ts.ImportDeclaration) => {
+      let moduleSpecifier = node.moduleSpecifier;
+      return {
+        specifierText: moduleSpecifier.getText().slice(1, -1),
+        pos: moduleSpecifier.pos,
+        end: moduleSpecifier.end
+      };
+    });
+}
+
+/** 
+ * @return a boolean value after it searches for a barrel (index.ts by convention) in a given path
+ * 
+ * @param filePath
+ * @param isDir boolean value to declare whether filePath is a directory or not. 
+ */
+export function hasIndexFile(filePath: string, isDir: boolean): Boolean {
```

Comment:
In general, boolean parameters are a bad idea in a method signature. They're hard to read and generally lead to poor APIs.

In this case, perhaps the caller should be responsible for providing only a directory name, instead of it being `hasIndexFile`'s responsibility to resolve the directory.